### ISSUE TITLE
Add time series examples

### ISSUE THREAD

**Issue by AndrewLister-STFC** [fitbenchmarking/fitbenchmarking#829]
Data assimilation problems can be given as time series data where the i+1'th x is equal to the i'th y. This should be quite straight forward to add support for.

In addition to this, time series data is about predicting future steps and will need plots that can show deviation from the time series by witholding some portion of the data.

E.g. For a simple example of doubling each time step the data may look like:
 
```
# X
1
2
4
8
```

In practice the function will be much more complicated (so it can't be reduced to a function in the line number!)

- [ ] Enable time series data in fitbenchmarking problems
- [ ] Add in some option for holding back a proportion of the data
- [ ] Generate plots that show a prediction (based on the optimisation) vs the actual result

**Comment by jess-farmer** [comment]
@AndrewLister-STFC to triage

**Comment by AndrewLister-STFC** [comment]
I don't think we have a use case for this at present? Suggest closing as not planned/wont fix
